lib/bolt/application.rb

```diff
@@ -0,0 +1,612 @@
+        list_guides(guides, outputter)
+      end
+      true
+    end
+
+    # Show inventory information.
+    #
+    # @param targets [Array[String], NilClass] The targets to show.
+    # @return [Hash]
+    #
+    def inventory_show(targets)
+      targets = group_targets_by_source(targets || ['all'])
+
+      {
+        adhoc: {
+          count: targets[:adhoc].count,
+          targets: targets[:adhoc].map(&:detail)
+        },
+        inventory: {
+          count: targets[:inventory].count,
+          targets: targets[:inventory].map(&:detail),
+          file: (inventory.source || config.default_inventoryfile).to_s,
+          default: config.default_inventoryfile.to_s
+        },
+        targets: targets.values.flatten.map(&:detail),
+        count: targets.values.flatten.count
+      }
+    end
+
+    # TODO: This should probably be separate #lookup and #plan_lookup methods.
+    #
+    # Lookup a value with Hiera.
+    #
+    # @param key [String] The key to look up in the hierarchy.
+    # @param targets [Array[String]] The targets to use as context.
+    # @param plan_hierarchy [Boolean] Lookup values using Hiera's plan_hierarchy.
+    # @param vars [Hash] Variables to set in the scope.
+    # @return [Bolt::ResultSet, String] The result of the lookup.
+    #
+    def lookup(key, targets, plan_hierarchy: false, vars: {})
+      if plan_hierarchy
+        pal.plan_hierarchy_lookup(key, plan_vars: vars)
+      else
+        executor.publish_event(type: :plan_start, plan: nil)
+
+        with_benchmark do
+          pal.lookup(key,
+                     inventory.get_targets(targets),
+                     inventory,
+                     executor,
+                     plan_vars: vars)
+        end
+      end
+    end
+
+    # Add a new module to the project.
+    #
+    # @param name [String] The name of the module to add.
+    # @param outputter [Bolt::Outputter] An outputter instance.
+    # @return [Boolean]
+    #
+    def module_add(name, outputter)
+      assert_project_file(config.project)
+
+      installer = Bolt::ModuleInstaller.new(outputter, pal)
+
+      installer.add(name,
+                    config.project.modules,
+                    config.project.puppetfile,
+                    config.project.managed_moduledir,
+                    config.project.project_file,
+                    config.module_install)
+    end
+
+    # Generate Puppet data types from project modules.
+    #
+    # @return [Boolean]
+    #
+    def module_generate_types
+      pal.generate_types(cache: true)
+    end
+
+    # Install the project's modules.
+    #
+    # @param outputter [Bolt::Outputter] An outputter instance.
+    # @param force [Boolean] Forcibly install modules.
+    # @param resolve [Boolean] Resolve module dependencies.
+    # @return [Boolean]
+    #
+    def module_install(outputter, force: false, resolve: true)
+      assert_project_file(config.project)
+
+      if config.project.modules.empty? && force
```

Comment:
Why did this get changed from `resolve` to `force`? It seems like this should still get skipped if resolve is false, and not be based on `force`...

Reply:
Sorry, the previous statement for reference:
```
if project.modules.empty? && resolve != false
```

Reply:
Derp. The logic needed updating since we don't need to account for `resolve` being `nil` anymore, but I accidentally switched it to `force`. 🤦